tests/test_ensemble.py

```diff
@@ -0,0 +1,381 @@
+        self.assertDictEqual(ens.mu.todict(), {'A':0.1,'B':0.2})
+        self.assertDictEqual(ens.N.todict(), {'A':None,'B':None})
+        self.assertDictEqual(ens.constant, {'P':True,
+                                            'V':False,
+                                            'mu':{'A':True,'B':True},
+                                            'N':{'A':False,'B':False}})
+        self.assertAlmostEqual(ens.beta, 0.1)
+
+        v_obj = relentless.Cube(L=3.0)
+
+        #V and N set
+        ens = relentless.Ensemble(T=20, V=v_obj, N={'A':1,'B':2})
+        self.assertCountEqual(ens.types, ('A','B'))
+        self.assertCountEqual(ens.rdf.pairs, (('A','B'),('A','A'),('B','B')))
+        self.assertAlmostEqual(ens.kB, 1.0)
+        self.assertAlmostEqual(ens.T, 20)
+        self.assertEqual(ens.P, None)
+        self.assertAlmostEqual(ens.V.volume, 27.0)
+        self.assertDictEqual(ens.mu.todict(), {'A':None,'B':None})
+        self.assertDictEqual(ens.N.todict(), {'A':1,'B':2})
+        self.assertDictEqual(ens.constant, {'P':False,
+                                            'V':True,
+                                            'mu':{'A':False,'B':False},
+                                            'N':{'A':True,'B':True}})
+        self.assertAlmostEqual(ens.beta, 0.05)
+
+        #mu and N used for one type each
+        ens = relentless.Ensemble(T=100, V=v_obj, mu={'A':0.1}, N={'B':2})
+        self.assertCountEqual(ens.types, ('A','B'))
+        self.assertCountEqual(ens.rdf.pairs, (('A','B'),('A','A'),('B','B')))
+        self.assertAlmostEqual(ens.kB, 1.0)
+        self.assertAlmostEqual(ens.T, 100)
+        self.assertEqual(ens.P, None)
+        self.assertAlmostEqual(ens.V.volume, 27.0)
+        self.assertDictEqual(ens.mu.todict(), {'A':0.1,'B':None})
+        self.assertDictEqual(ens.N.todict(), {'A':None,'B':2})
+        self.assertDictEqual(ens.constant, {'P':False,
+                                            'V':True,
+                                            'mu':{'A':True,'B':False},
+                                            'N':{'A':False,'B':True}})
+        self.assertAlmostEqual(ens.beta, 0.01)
+
+        #test creation with single type
+        ens = relentless.Ensemble(T=100, V=v_obj, mu={'A':0.1})
+        self.assertCountEqual(ens.types, ('A',))
+        self.assertCountEqual(ens.rdf.pairs, (('A','A'),))
+        self.assertAlmostEqual(ens.kB, 1.0)
+        self.assertAlmostEqual(ens.T, 100)
+        self.assertEqual(ens.P, None)
+        self.assertAlmostEqual(ens.V.volume, 27.0)
+        self.assertDictEqual(ens.mu.todict(), {'A':0.1})
+        self.assertDictEqual(ens.N.todict(), {'A':None})
+        self.assertDictEqual(ens.constant, {'P':False,
+                                            'V':True,
+                                            'mu':{'A':True},
+                                            'N':{'A':False}})
+        self.assertAlmostEqual(ens.beta, 0.01)
```

Comment:
In one of these tests, could set Boltzmann's constant to something other than 1.0 to make sure that beta is using it properly? For example, its physical value is 1.38e-23 J/K but you can just use some other number like 2.0.